BusBooking.cls

```diff
@@ -0,0 +1,140 @@
+       HTTPResponse resp = http.send(req);       
+       JSONParser parser = JSON.createParser(resp.getBody()); //Json parser to parse the json string obtained from url
+       Double latitude = null;
+       Double longitude = null;
+       while (parser.nextToken() != null)
+       {
+       if ((parser.getCurrentToken() == JSONToken.FIELD_NAME)  && (parser.getText() == 'location'))
+       {
+       // Jump to latitude and longtitude fields.
+       parser.nextToken();
+       while (parser.nextToken() != JSONToken.END_OBJECT)
+       {
+       String text = parser.getText();
+       parser.nextToken();
+       if (text == 'lat') latitude = parser.getDoubleValue();
+       else if (text == 'lng') longitude = parser.getDoubleValue();
+          }
+          } 
+          }
+       String query = 'SELECT School__c FROM Stop__c WHERE';
+       query+= ' DISTANCE(Location__c, GEOLOCATION('+latitude+','+longitude+'),\'mi\') < 10 ';
+       query+='ORDER BY DISTANCE(Location__c, GEOLOCATION('+latitude+','+longitude+'),\'mi\') LIMIT 3';       
+       if(latitude != null && longitude!= null){
+       List<Stop__c> stoplist = Database.query(query);            
+       Set<id> setstopid = new Set<Id>();             
+       for(Stop__C obj:stoplist){
+       setstopid.add(obj.id);
+          }             
+       lstroutestop=[SELECT Route__c,Route__r.name,Stop__r.name,Route__r.From__c,Route__r.To__c FROM Route_Stop__c where Stop__c in :setstopid ];
+       for(integer i=0;i<lstroutestop.size();i++){
+           if(lstroutestop.get(i).Stop__r.name != selectedstops)
+               lstroutestop.remove(i);
+       }     
+          }                    
+         }
+           
+             
+     /** Get availability method to display the availability of seats**/  
+      public void getavailability(){
+        System.debug('rcordid'+recid);
+        Vehicle__c vehicleobj = new Vehicle__c();
+        vehroute=[SELECT VehicleJunction__r.name,VehicleJunction__r.Capacity__c,VehicleJunction__c,RouteJunction__r.name,RouteJunction__c FROM Vehicle_Route__c WHERE   RouteJunction__c= :recid limit 1 ];
+        vehicleobj =[Select Capacity__c from Vehicle__c where id= :vehroute.VehicleJunction__c];
+        System.debug('vehicle list value is'+vehicleobj);                                  
+        doub = vehicleobj.Capacity__c;
+        System.debug('doub value is'+doub);
+        doubcapacity =doub - nooftickets;
+        vehicleobj.Capacity__c = doubcapacity;
+        System.debug('new capacity value is'+doubcapacity);
+        if(doubcapacity >0)
+         {
+           availablesection=true;   
+           notavailablesection=false;
+         }
+         else
+         {
+           notavailablesection=true;    
+           availablesection=false;
+          }               
+         }
+              
+              
+       /** Gotobookingpage method to redirect the user to the booking page**/     
+       public PageReference gotoBookingpage(){
```

Comment:
Remove this method and instead of manual click on available link direct redirect user to this page is booking is available.